sdk/cosmosdb/cosmos/src/queryIterator.ts

```diff
@@ -266,15 +270,53 @@ export class QueryIterator<T> {
     );
   }
 
-  private async createPipelinedExecutionContext(): Promise<void> {
+  private async createExecutionContext(diagnosticNode?: DiagnosticNodeInternal): Promise<void> {
     const queryPlanResponse = await this.queryPlanPromise;
 
     // We always coerce queryPlanPromise to resolved. So if it errored, we need to manually inspect the resolved value
     if (queryPlanResponse instanceof Error) {
       throw queryPlanResponse;
     }
 
-    const queryPlan = queryPlanResponse.result;
+    const queryPlan: PartitionedQueryExecutionInfo = queryPlanResponse.result;
+    if (queryPlan.hybridSearchQueryInfo && queryPlan.hybridSearchQueryInfo !== null) {
+      await this.createHybridQueryExecutionContext(queryPlan, diagnosticNode);
+    } else {
+      await this.createPipelinedExecutionContext(queryPlan);
+    }
+  }
+
+  private async createHybridQueryExecutionContext(
+    queryPlan: PartitionedQueryExecutionInfo,
+    diagnosticNode?: DiagnosticNodeInternal,
+  ): Promise<void> {
+    const allPartitionKeyRanges = (
```

Comment:
can be moved to global statistics query creation as this is needed just for global statics or FTS queries